x-pack/platform/plugins/shared/onechat/server/services/tools/persisted/tool_types/esql/esql_tool_type.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ToolType } from '@kbn/onechat-common';
+import type { EsqlToolConfig } from '@kbn/onechat-common';
+import type { PersistedToolTypeDefinition } from '../types';
+import { toToolDefinition } from './to_tool_definition';
+import { validateConfig } from './validate_configuration';
+import { configurationSchema, configurationUpdateSchema } from './schemas';
+
+export const createEsqlToolType = (): PersistedToolTypeDefinition<EsqlToolConfig> => {
+  return {
+    toolType: ToolType.esql,
+    toToolDefinition,
+    createSchema: configurationSchema,
+    updateSchema: configurationUpdateSchema,
+    validateForCreate: async (config) => {
+      await validateConfig(config);
+      return config;
+    },
```

Comment:
We now have two tools "sources" (yes, naming is hard) - one for built-in and one for persisted tools. And the persisted tool source handles multiple tool types (esql and index_search).

This is still now perfect and will require a few more iterations (e.g. if we want to let solution statically register tools of persisted types), but it's still one step closer to the end goal, given we have a single persistence client for all persisted tools.